### PR TITLE
Add /get_stray_cats API for accessing unknown CATs

### PR DESCRIPTION
**Pull request by ytx1991**
This is a draft version. Unit tests will be added after there is no concern about the implementation.

### CODE REVIEW

**Comment by ojura** [comment]
@ytx1991 great to see this incoming. I have an approved related PR waiting to be merged and is going to clash a bit with yours:

https://github.com/Chia-Network/chia-blockchain/pull/10308

Is there a chance you could rebase this on top of my branch?

**Comment by ytx1991** [comment]
> @ytx1991 great to see this incoming. I have an approved related PR waiting to be merged which is going to clash a bit with yours:
> 
> #10308
> 
> Is there a chance you could rebase this on top of my branch? An important part of #10308 is centralizing the name generation for new unknown CATs; I see you're adding another hardcoded name (`Unknown`) in this PR.

Thanks for the notification. I will, but I'm not very familiar with the git PR, is my PR blocked your merge?

**Comment by ojura** [comment]
> Thanks for the notification. I will, but I'm not very familiar with the git PR, is my PR blocked your merge?

There is going to be a conflict, since both our PRs touch the same stuff, like that piece of code "if cat tail is in trusted list: create wallet for it". 

I also worked to remove hardcoded names for new unknown CATs throughout the codebase and to have one centralized place where the way to generate them is defined, and that is `CATWallet.default_wallet_name_for_unknown_cat()`. You'll be going against that by introducing a new hardcoded value, `Unknown`. So I'd appreciate if you have a look at my PR and get familiarized with the changes :)

**Comment by ytx1991** [comment]
> > Thanks for the notification. I will, but I'm not very familiar with the git PR, is my PR blocked your merge?
> 
> @ytx1991 there is going to be a conflict, since both our PRs touch the same stuff, like that piece of code "if cat tail is in trusted list: create wallet for it".
> 
> I also worked to remove hardcoded names for new unknown CATs throughout the codebase and to have one centralized place where the way to generate them is defined, and that is `CATWallet.default_wallet_name_for_unknown_cat()`. You'll be going against that by introducing a new hardcoded value, `Unknown`. So I'd appreciate if you have a look at my PR and get familiarized with the (approved!) changes :)

I read the code, I just want to wait your change merged to the main branch then rebase. If you cannot merge your change because mine, let me know.

File: chia/wallet/wallet_state_manager.py [hunk 1]
```diff
@@ -603,6 +604,14 @@ async def fetch_parent_and_check_for_cat(
...
                     wallet_type = WalletType(cat_wallet.type())
                     self.state_changed("wallet_created")
+                else:
+                    # Found unacknowledged CAT, save it in the database.
+                    await self.interested_store.add_unacknowledged_token(
+                        asset_id,
+                        CATWallet.default_wallet_name_for_unknown_cat(asset_id.hex()),
+                        parent_coin_state.spent_height,
+                        parent_coin_state.coin.puzzle_hash,
+                    )
```
Comment (paninaro) [inline chia/wallet/wallet_state_manager.py:614]:
```suggestion
                    )
                    self.state_changed("added_unacknowledged_token")
```

**Review by paninaro — APPROVED** [review]
The exposed RPC name doesn't sound quite right, but I don't have a great alternative to suggest.

Can we rename the RPC (just the RPC name) to `get_stray_cats`? @ytx1991

File: chia/rpc/wallet_rpc_api.py [hunk 2]
```diff
@@ -832,6 +833,16 @@ async def cat_get_name(self, request):
         name: str = await wallet.get_name()
         return {"wallet_id": wallet_id, "name": name}
 
+    async def cat_get_unacknowledged(self, request):
```
Comment (paninaro) [inline chia/rpc/wallet_rpc_api.py:836]:
```suggestion
    async def get_stray_cats(self, request):
```

File: chia/rpc/wallet_rpc_api.py [hunk 3]
```diff
@@ -90,6 +90,7 @@ def get_routes(self) -> Dict[str, Callable]:
             "/cat_set_name": self.cat_set_name,
             "/cat_asset_id_to_name": self.cat_asset_id_to_name,
             "/cat_get_name": self.cat_get_name,
+            "/cat_get_unacknowledged": self.cat_get_unacknowledged,
```
Comment (paninaro) [inline chia/rpc/wallet_rpc_api.py:93]:
```suggestion
            "/get_stray_cats": self.get_stray_cats,
```

File: chia/rpc/wallet_rpc_client.py [hunk 4]
```diff
@@ -351,6 +351,10 @@ async def get_cat_asset_id(self, wallet_id: str) -> bytes:
         }
         return bytes.fromhex((await self.fetch("cat_get_asset_id", request))["asset_id"])
 
+    async def get_unacknowledged_cats(self) -> Dict:
+        response = await self.fetch("cat_get_unacknowledged", {})
```
Comment (paninaro) [inline chia/rpc/wallet_rpc_client.py:355]:
```suggestion
        response = await self.fetch("get_stray_cats", {})
```

**Review by paninaro — CHANGES_REQUESTED** [review]
Suggesting the RPC name change to `get_stray_cats` since that's what people liked the most.

File: chia/wallet/wallet_state_manager.py [hunk 5]
```diff
@@ -603,6 +604,15 @@ async def fetch_parent_and_check_for_cat(
...
+                    self.state_changed("added_unacknowledged_token")
```
Comment (paninaro) [inline chia/wallet/wallet_state_manager.py:615]:
```suggestion
                    self.state_changed("added_stray_cat")
```

**Review by paninaro — APPROVED** [review]

**Pull request merged**